packages/zowe-explorer-api/src/vscode/ZoweVsCodeExtension.ts

```diff
@@ -234,12 +236,15 @@ export class ZoweVsCodeExtension {
             });
             await (zeRegister?.getCommonApi(serviceProfile).logout ?? Logout.apimlLogout)(updSession);
 
-            const connOk = serviceProfile.profile.host === baseProfile.profile.host && serviceProfile.profile.port === baseProfile.profile.port;
-            if (connOk) {
-                await cache.updateBaseProfileFileLogout(baseProfile);
-            } else {
-                await cache.updateBaseProfileFileLogout(serviceProfile);
-            }
+            const connOk =
+                serviceProfile.profile.host === baseProfile.profile.host &&
+                serviceProfile.profile.port === baseProfile.profile.port &&
+                !serviceProfile.name.startsWith(baseProfile.name + ".");
+            await cache.updateBaseProfileFileLogout(connOk ? baseProfile : serviceProfile);
```

Comment:
I'm guessing that for logout purposes it doesn't matter if the service profile is nested inside an actual base profile? 😋